From: Ben Hutchings <ben@decadent.org.uk>
Date: Sat, 07 May 2016 21:19:08 +0100
Subject: PCI: Avoid ABI changes in 3.16.y
Forwarded: not-needed

Move the added bitfield pci_dev::no_64bit_msi to the end of its
bitfield group and hide it from genksyms.

Also hide pci_dev::non_compliant_bars from genksyms.

--- a/include/linux/pci.h
+++ b/include/linux/pci.h
@@ -335,7 +335,6 @@ struct pci_dev {
 	unsigned int	is_added:1;
 	unsigned int	is_busmaster:1; /* device is busmaster */
 	unsigned int	no_msi:1;	/* device may not use msi */
-	unsigned int	no_64bit_msi:1; /* device may only use 32-bit MSIs */
 	unsigned int	block_cfg_access:1;	/* config space access is blocked */
 	unsigned int	broken_parity_status:1;	/* Device generates false positive parity */
 	unsigned int	irq_reroute_variant:2;	/* device needs IRQ rerouting variant */
@@ -353,7 +352,11 @@ struct pci_dev {
 	unsigned int	__aer_firmware_first:1;
 	unsigned int	broken_intx_masking:1;
 	unsigned int	io_window_1k:1;	/* Intel P2P bridge 1K I/O windows */
+#ifndef __GENKSYMS__
+	unsigned int	no_64bit_msi:1; /* device may only use 32-bit MSIs */
 	unsigned int	non_compliant_bars:1;	/* broken BARs; ignore them */
+	/* 7 bits spare */
+#endif
 	pci_dev_flags_t dev_flags;
 	atomic_t	enable_cnt;	/* pci_enable_device has been called */
 
